resource-managers/kubernetes/docker/src/main/dockerfiles/spark/Dockerfile

```diff
@@ -51,4 +51,11 @@ ENV SPARK_HOME /opt/spark
 
 WORKDIR /opt/spark/work-dir
 
+# add the following two lines before ENTRYPOINT
+RUN rm $SPARK_HOME/jars/kubernetes-client-*.jar
+RUN rm $SPARK_HOME/jars/guava-14.0.1.jar
+ADD https://repo1.maven.org/maven2/io/fabric8/kubernetes-client/4.4.2/kubernetes-client-4.4.2.jar $SPARK_HOME/jars
+ADD https://repo1.maven.org/maven2/com/google/guava/guava/23.0/guava-23.0.jar $SPARK_HOME/jars
```

Comment:
@kkolman

Reply:
Spark 2.4.2 needs a Guava that is <= 16.0 .

Reply:
same answer I gave above.

Reply:
@peri4n wasnt the guava version (and the conflicts) requirement of the hadoop-2.6.5 deps ?

because as far as i can see
guava is a provided dependency, so you "should be able to swap in whichever you like"
https://mvnrepository.com/artifact/org.apache.spark/spark-core_2.12/2.4.4

and the whichever you like would be the one coming from the "hadoop" version

but of course dependency hell is purely theoriticaly right until it fails in production...